docs/tutorial/reading_writing.md

```diff
@@ -0,0 +1,336 @@
+# Reading and writing
+
+In this section, we will guide you through the process of opening and extracting information from files using our simple, intuitive API. Whether you are working with text, binary, or other formats, you will find the tools you need to seamlessly integrate file reading into your Scala project. Let us dive in and unlock the power of data stored in files!
+
+**Important:** To avoid tutorial hell, we recommend doing a really small project while you read this tutorial to place acquired knowledge into practice. It does not have to be fancy, but it should tap into your curiosity and challenge you just a little.
+
+## Reading and printing a file
+
+Let's say you want to load the contents of a file in memory...
+
+The first thing you need to do is import the `read` method and the `Path` type:
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala
+import shellfish.Path
+import shellfish.syntax.path.*
+```
+
+@:choice(static)
+
+```scala
+import shellfish.Path
+import shellfish.FilesOs
+```
+
+@:@
+
+Next, define the `run` function to execute the `IO`. To do that, you need to extend your application with a `cats.effect.IOApp`, let us name it `App`:
+
+```scala mdoc:compile-only
+import cats.effect.{IO, IOApp}
+
+object App extends IOApp.Simple:
+
+  def run: IO[Unit] = ???
+
+end App
+```
+
+Now we can start using the library! First, create a `Path` containing the path to the file you want to read:
+
+```scala mdoc:compile-only
+import fs2.io.file.Path // TODO: delete this and use the Shellfish's one
+val path = Path("testdata/readme.txt")
+```
+
+And use the `read` function to load the file in memory as a string:
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:fail
+import cats.effect.{IO, IOApp}
+
+object App extends IOApp.Simple:
+
+  val path = Path("testdata/readme.txt")
+
+  def run: IO[Unit] = path.read
+
+end App
+```
+
+@:choice(static)
+
+```scala mdoc:fail
+import cats.effect.{IO, IOApp}
+
+object App extends IOApp.Simple:
+
+  val path = Path("testdata/readme.txt")
+
+  def run: IO[Unit] = FilesOs.read(path)
+
+end App
+```
+
+@:@
+
+Oops! We got an error saying that the `run` function accepts an `IO[Unit]`, but the `read` function returns an `IO[String]`. This happens because we are not doing anything with the string loaded and therefore not returning `IO[Unit]`. Fortunately, because `IO` is a monad, we can sequence the value obtained by evaluating the `read` computation with another that prints things to the console (and thus returns `IO[Unit]`!). This is achieved by using the `flatMap` function as follows:
```

Comment:
```suggestion
Oops! We got an error saying that the `run` function accepts an `IO[Unit]`, but the `read` function returns an `IO[String]`. This happens because we are not doing anything with the string and therefore not returning `IO[Unit]`. Let's fix that by sequencing the value obtained from `read` with another that prints it to the console (and thus returns `IO[Unit]`!). This is achieved by using the `flatMap` function as follows:
```

Scary  m-word